sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SimpleFileDiagnosticsProvider.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.spark.diagnostics.{DiagnosticsProvider, ILogger, SimpleDiagnosticsSlf4jLogger}
+
+import java.io.File
+import scala.collection.concurrent.TrieMap
+
+// only when diagnostics enabled,
+// - logs each individual writes success and/or failures with id,pk
+// - logs each documentServiceRequest and documentServiceResponse
+private[spark] class SimpleFileDiagnosticsProvider extends DiagnosticsProvider {
+  override def getLogger(classType: Class[_]): ILogger =
+    SimpleFileDiagnosticsProvider.getOrCreateSingletonLoggerInstance(classType)
+}
+
+private[spark] object SimpleFileDiagnosticsProvider {
+  val singletonInstances: TrieMap[Class[_], SimpleFileDiagnosticsSlf4jLogger] =
+    new TrieMap[Class[_], SimpleFileDiagnosticsSlf4jLogger]()
+
+  private val folderName = System.getProperty("java.io.tmpdir") + "/SimpleFileDiagnostics"
+  private val folder = new File(folderName)
+
+  def getOrCreateSingletonLoggerInstance(classType: Class[_]): SimpleFileDiagnosticsSlf4jLogger = {
+    if (!folder.exists()) {
+      folder.mkdirs()
```

Comment:
there are some APIs for ensuring deletion of the temporary files/folders if the JVM exists:

`Files.createTempDirectory`
https://stackoverflow.com/questions/54946535/how-to-create-temporary-directory-in-scala-unit-tests

Reply:
the createTempFile/createTempDirectory also isn't really suitable because they generate random File/Folder names - which causes trouble when trying to merge logs across executor(s)/master VMs. But I added some more robust cleanup handling - thanks for catching this!